shoplist/src/main/java/com/netguru/codereview/ui/MainViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.netguru.codereview.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.netguru.codereview.network.ShopListApiMock
+import com.netguru.codereview.network.ShopListRepository
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+
+    private val shopListRepository = ShopListRepository(ShopListApiMock())
+
+    val shopLists = MutableLiveData<List<Pair<ShopListResponse, List<ShopListItemResponse>>>>()
+    private val eventLiveData = MutableLiveData<String>()
+
+    init {
+        viewModelScope.launch {
+            val lists = shopListRepository.getShopLists()
+            val data = mutableListOf<Pair<ShopListResponse, List<ShopListItemResponse>>>()
+            for (list in lists) {
+                val items = shopListRepository.getShopListItems(list.list_id)
+                data.add(list to items)
+            }
+            shopLists.postValue(data)
+        }
+        getUpdateEvents()
+    }
+
+    fun events(): LiveData<String> = eventLiveData
+    private fun getUpdateEvents() {
+        GlobalScope.launch {
```

Comment:
1. Consider changing the accessor method name from events to getUpdateEvents or changing it from method to public variable, because events is a collective noun and a function performs an action regardless of the return type. Try naming is to a verb. In this case, turning it to a public variable as a LiveData accessor sounds better
2. Consider renaming the private function name from getUpdateEvents to loadUpdateEvents or observeUpdateEvents as we are not requesting a single event and ending it but we are actively listening to updates on the flow of the event.